src/components/tasks/card/index.tsx

```diff
@@ -41,31 +41,57 @@ const Card: FC<Props> = ({
     const timeLeft = localEndsOn.valueOf() - currentDate.valueOf();
     return !statusNotOverDueList.includes(cardDetails.status) && timeLeft <= 0;
   }
-
+  
   function stripHtml(html: string) {
     const tmp = document.createElement('DIV');
     tmp.innerHTML = html;
     return tmp.textContent || tmp.innerText || '';
   }
-
+  
   function handleChange(event: any, changedProperty: keyof typeof cardDetails) {
     if (event.key === 'Enter') {
       const toChange: any = cardDetails;
       toChange[changedProperty] = stripHtml(event.target.innerHTML);
-
+      
       if (changedProperty === 'endsOn' || changedProperty === 'startedOn') {
         const toTimeStamp = new Date(`${toChange[changedProperty]}`).getTime() / 1000;
         toChange[changedProperty] = toTimeStamp;
       }
-
+      
       onContentChange(toChange.id, {
         [changedProperty]: toChange[changedProperty],
       });
     }
   }
+  
+  function handleProgressColor(percentCompleted: number, startedOn: string, endsOn: string): string {
+    const startDate = moment(new Date(parseInt(startedOn, 10) * 1000))
```

Comment:
can we please make the logic of new Date a little generic so rather than we writing it again and again we have a function that does the work for us & can be called where needed